app/actions/agentrelationship/request/AppointAgentSessionRequest.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package actions.agentrelationship.request
+
+import actions.requests.CcaWrappedRequest
+import models.propertyrepresentation.AppointNewAgentSession
+import models.{Accounts, DetailedIndividualAccount, GroupAccount, LinkingSession}
+import play.api.mvc.{Request, WrappedRequest}
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+case class AppointAgentSessionRequest[A](
+      sessionData: AppointNewAgentSession,
+      organisationId: Long,
```

Comment:
organisationId comes from the groupAccount correct? just wondering if instead of having this as part of the constructor, we have a helper method inside the class that gets the organisationId out.